frontend/src/components/common/formQuestions/AddQuestionModal.tsx

```diff
@@ -0,0 +1,300 @@
+    option: string;
+  }
+
+  const key: keyof Option = "option";
+
+  const [options, setOptions] = useState<Array<Option>>([{ option: "" }]);
+
+  const handleFormChange = (
+    index: number,
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const data = [...options];
+    data[index][key] = e.target.value;
+    setOptions(data);
+    const optionsArray = data.map((option) => option.option);
+    setOptionsArray(optionsArray);
+  };
+
+  const removeField = (index: number) => {
+    const data = [...options];
+    data.splice(index, 1);
+    setOptions(data);
+    const optionsArray = data.map((option) => option.option);
+    setOptionsArray(optionsArray);
+  };
+
+  const addInputField = () => {
+    const newField = { option: "" };
+    setOptions([...options, newField]);
+  };
+
+  return (
+    <VStack minW="100%" key="options_stack">
+      {options.map((option, index) => {
+        return (
+          <HStack
+            key={`${option}_${index}_stack`}
+            minW="100%"
+            align="start"
+            alignItems="center"
+          >
+            {questionType === "MultipleChoice" && (
+              <Radio size="lg" key={`${option}_${index}_radio`} isReadOnly />
+            )}
+
+            {questionType === "Multiselect" && (
+              <Checkbox
+                isReadOnly
+                key={`${option}_${index}_checkbox`}
+                size="lg"
+              />
+            )}
+
+            <Input
+              key={`${option}_${index}_input`}
+              onChange={(e) => handleFormChange(index, e)}
+              value={option.option}
+            />
+
+            <IconButton
+              icon={<DeleteIcon />}
+              aria-label="remove option"
+              variant="ghost"
+              key={`${option}_${index}_delete`}
+              maxW="15px"
+              onClick={() => removeField(index)}
+              isDisabled={options.length <= 1}
+            />
+          </HStack>
+        );
+      })}
+      <IconButton
+        aria-label="add new option"
+        icon={<AddIcon />}
+        minW="100%"
+        maxH="30px"
+        onClick={addInputField}
+      />
+    </VStack>
+  );
+};
+
+type AddQuestionModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onSave: (formQuestion: CreateFormQuestion) => void;
+};
+
+const AddQuestionModal = ({
+  isOpen,
+  onClose,
+  onSave,
+}: AddQuestionModalProps): React.ReactElement => {
+  const [question, setQuestion] = useState<string>("");
+  const [questionCategory, setQuestionCategory] = useState<string>(
+    "PersonalInfo",
+  );
+  const [questionType, setQuestionType] = useState<string>("Text");
+  const [questionDescription, setQuestionDescription] = useState<string>("");
+  const [isRequiredQuestion, setIsRequiredQuestion] = useState<boolean>(false);
+  const [questionOptions, setQuestionOptions] = useState<Array<string>>([]);
+
+  const [isQuestionInvalid, setIsQuestionInvalid] = useState<boolean>(false);
```

Comment:
I'm a little unclear on this one - why does it have to be a state variable? If the page renders each time, can't we just use a constant and re-evaluate `const isQuestionEmpty = question === "";`

Reply:
I think I did it this way because the value of the question state starts off as "" so the form would start off as invalid. Using state, we have closer control of when to make it invalid

Reply:
Can we rename it to `isQuestionEmpty` then?